docs/spec_draft.md

```diff
@@ -3080,6 +3082,81 @@ More formally, for each element `x`: `real(x) = is_complex(x) ? x.real : x`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.recv
+
+### Semantics
+
+Receives data into a partition from a [stablehlo.send](#stablehlosend)
+instruction, in another partition, that shares the same channel handle pair
+{`channel_id`, `channel_type`}.
+
+`stablehlo.recv`, paired with the corresponding [stablehlo.send](#stablehlosend)
+instruction, represents synchronous communication. However, in order to enable
+asynchronous data transfers, it is decomposed into two HLO instructions
+(`Recv` and `RecvDone`). Similarly, the paired
+[stablehlo.send](#stablehlosend) is internally decomposed into two HLO
+instructions (`Send` and `SendDone`). [todo](https://github.com/openxla/stablehlo/issues/642)
+
+These 4 instructions, each associated with a particular channel, has the
+following semantics:
+
+  * **Recv**: Initiator of the channel by allocating the receive buffer and
```

Comment:
Initiates the channel, allocates the receive buffer and notifies the sender to start sending data.

Reply:
keeping the language and form consistent with the next line.

Reply:
To me it seems just a matter of preference for the receiver to say "I am ready to receive" Vs "Feel free to send the data". 
The semantics of this notification is taken from an internal doc. I would prefer keeping it the same.